src/robot/running/namespace.py

```diff
@@ -297,21 +298,23 @@ def _get_runner(self, name):
             runner = self._get_explicit_runner(name)
         if not runner:
             runner = self._get_implicit_runner(name)
-        if not runner:
-            runner = self._get_bdd_style_runner(name, self.languages.bdd_prefixes)
+        # handle bdd prefixes, even when swallowed by a leading variable
+        if not runner or not search_variable(runner.keyword.name).before:
```

Comment:
First getting a runner and then trying to get it again if the keyword name has a variable with something before it is a bit complicated way to implement this. It would be a lot simpler to just first try to find a BDD-style runner. I tested that this seems to work:
```diff
--- a/src/robot/running/namespace.py
+++ b/src/robot/running/namespace.py
@@ -286,13 +286,13 @@ class KeywordStore:
             raise DataError('Keyword name cannot be empty.')
         if not is_string(name):
             raise DataError('Keyword name must be a string.')
-        runner = self._get_runner_from_suite_file(name)
+        runner = self._get_bdd_style_runner(name, self.languages.bdd_prefixes)
+        if not runner:
+            runner = self._get_runner_from_suite_file(name)
         if not runner and '.' in name:
             runner = self._get_explicit_runner(name)
         if not runner:
             runner = self._get_implicit_runner(name)
-        if not runner:
-            runner = self._get_bdd_style_runner(name, self.languages.bdd_prefixes)
         return runner
```
The main benefit would be that the algorithm stays consistent. We always try to find a `runner` and if it fails, we try again using a different approach.

The only drawback I see is that the current approach to go through all possible BDD-prefixes in `_get_bdd_style_runner` can be a bit slow. Splitting the used keyword name from spaces and then trying all possible combinations one-by-one is a bit stupid when actual prefixes we have contain maximum two parts. I doubt that has a real performance effect, but I have some ideas how to easily enhance that.

Reply:
Good suggestion. So far I haven't found a downside to the simpler solution, so I took it exactly as suggested.